Modules/_csv.c

```diff
@@ -603,6 +603,34 @@ Dialect_traverse(DialectObj *self, visitproc visit, void *arg)
     return 0;
 }
 
+static PyObject *
+Dialect_str(DialectObj *self) {
+    Py_ssize_t pos = 0;
+    PyObject *key, *value;
+    PyObject *module = PyType_GetModule(Py_TYPE(self));
+    PyObject *dialects = get_csv_state(module)->dialects;
+
+    while (PyDict_Next(dialects, &pos, &key, &value)) {
+        if (value == (PyObject *)self) {
+            Py_INCREF(key);
+
+            return key;
+        }
+    }
+
+    return PyUnicode_FromString("unknown");
+}
+
+static PyObject *
+Dialect_repr(DialectObj *self) {
+    PyObject *str = Dialect_str(self);
```

Comment:
This can also return `NULL` (`PyUnicode_FromString` can fail), so we need to check for that.